server/app/controllers/ti/TrustedIntermediaryController.java

```diff
@@ -110,36 +113,30 @@ public Result updateDateOfBirth(Long tiId, Long applicantId, Http.Request reques
     if (trustedIntermediaryGroup.isEmpty()) {
       return notFound();
     }
-    if (!trustedIntermediaryGroup.get().id.equals(tiId)) {
-      return unauthorized();
-    }
-    Form<UpdateApplicantDOB> form =
-        formFactory.form(UpdateApplicantDOB.class).bindFromRequest(request);
-    if (form.hasErrors()) {
-      return redirectToDashboardWithUpdateDateOfBirthError(form.errors().get(0).message(), form);
-    }
-    if (Strings.isNullOrEmpty(form.get().getDob())) {
-      return redirectToDashboardWithUpdateDateOfBirthError("Date Of Birth is required.", form);
-    }
-
+    Form<UpdateApplicantDob> form =
+        formFactory.form(UpdateApplicantDob.class).bindFromRequest(request);
     try {
-      Applicant applicant =
-          userRepository.lookupApplicant(applicantId).toCompletableFuture().join().get();
-      applicant.getApplicantData().setDateOfBirth(form.get().getDob());
-      userRepository.updateApplicant(applicant);
-    } catch (DateOfBirthExistsException e) {
-      String trustedIntermediaryUrl = baseUrl + "/trustedIntermediaries";
-
-      return redirectToDashboardWithUpdateDateOfBirthError(
-          "Date Of birth for this applicant exists.  Cannot Update DOB. "
-              + " Direct applicant to sign in and go to"
-              + " "
-              + trustedIntermediaryUrl,
-          form);
+      tiService.checkFormForDobUpdate(form, applicantId);
+    } catch (IncorrectDateFormatException e) {
+      redirectToDashboardWithUpdateDateOfBirthError(e.getLocalizedMessage(), form);
```

Comment:
Generally I've found the codebase to be a bit inconsistent here, but one pattern we use rather than throwing exceptions for validation errors is to return an `ErrorAnd<THERETURNTYPE, CiviFormError>`. Then the caller can check if there are errors and perform similar logic to what you have here.

Probably for another time, but some of this validation logic could likely be embedded into the form itself.